tests/scripts/thread-cert/Cert_8_3_01_CommissionerPetition.py

```diff
@@ -0,0 +1,366 @@
+        #         response (LEAD_KA.rsp) to Commissioner:
+        #         CoAP Response Code
+        #             2.04 Changed
+        #         CoAP Payload
+        #             State TLV (value = Reject)
+        #
+        #         Leader sends a MLE Data Response to the network with the
+        #         following TLVs:
+        #             - Active Timestamp TLV
+        #             - Leader Data TLV
+        #             - Network Data TLV
+        #             - Source Address TLV
+        pkts.filter_ipv6_src_dst(LEADER_RLOC, COMMISSIONER_RLOC).\
+            filter_coap_ack(LEAD_KA_URI).\
+            filter(lambda p: {
+                              NM_STATE_TLV
+                              } == set(p.coap.tlv.type) and\
+                   p.thread_meshcop.tlv.state == MESHCOP_REJECT
+                   ).\
+           must_next()
+        pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_DATA_RESPONSE).\
+            filter(lambda p: {
+                              NETWORK_DATA_TLV,
+                              SOURCE_ADDRESS_TLV,
+                              ACTIVE_TIMESTAMP_TLV,
+                              LEADER_DATA_TLV
+                             } == set(p.mle.tlv.type) and\
+                             {
+                              NWD_COMMISSIONING_DATA_TLV
+                             } == set(p.thread_nwd.tlv.type) and\
+                             {
+                              NM_COMMISSIONER_SESSION_ID_TLV
+                             } == set(p.thread_meshcop.tlv.type) and\
+                   p.mle.tlv.leader_data.data_version >=
+                   (_dr_pkt2.mle.tlv.leader_data.data_version + 1) % 256 and\
+                   p.thread_nwd.tlv.stable == [0]
+                   ).\
+            must_next()
+
+        # Step 10: Commissioner sends a petition request (LEAD_PET.req) to Leader:
+        #          CoAP Request URI
+        #              CON POST coap://<L>:MM/c/lp
+        #          CoAP Payload
+        #              Commissioner ID TLV
+        pkts.filter_wpan_src64(COMMISSIONER).\
+            filter_ipv6_2dsts(LEADER_ALOC, LEADER_RLOC).\
+            filter_coap_request(LEAD_PET_URI).\
+            filter(lambda p: {
+                              NM_COMMISSIONER_ID_TLV
+                              } == set(p.coap.tlv.type)\
+                   ).\
+           must_next()
+
+        # Step 11: Leader accepts Commissioner to the network, sends a Leader Petition
+        #          Response (LEAD_PET.rsp) to Commissioner:
+        #          CoAP Response Code
+        #              2.04 Changed
+        #          CoAP Payload
+        #              State TLV (value = Accept)
+        #              Commissioner ID TLV (optional)
+        #              Commissioner Session ID TLV (contains higher Session ID number
```

Comment:
Here the description in Test Plan might not be appropriate -these two are marked as optional in Thread Specification and only appear when accept, and skip when reject.
May file a ticket to Thread Group?

Additionally, session id number seems not checked

Reply:
sure